pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -300,18 +300,22 @@ func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldOb
 	return framework.Queue, nil
 }
 
-// isResourceScaleDown checks whether the resource request of the modified pod is less than the original pod
-// for the resources requested by the pod we are trying to schedule.
-func (f *Fit) isResourceScaleDown(targetPod, originalOtherPod, modifiedOtherPod *v1.Pod) bool {
-	if modifiedOtherPod.Spec.NodeName == "" {
-		// no resource is freed up whatever the pod is modified.
+// isResourceScaleDown checks whether an update event may make the pod schedulable. Specifically:
+// - Returns true when an update event shows a scheduled pod's resource request got reduced.
+// - Returns true when an update event is for the unscheduled pod itself, and it shows the pod's resource request got reduced.
+func (f *Fit) isResourceScaleDown(targetPod, originalPod, modifiedPod *v1.Pod) bool {
+	samePod := modifiedPod.Namespace == targetPod.Namespace && modifiedPod.Name == targetPod.Name
+	scheduledPod := modifiedPod.Spec.NodeName != ""
+	if !(samePod) && (!scheduledPod) {
```

Comment:
Is this possible, based on my understanding, we should only verify the podRequests changed then a failed pod might be schedulable again.

Reply:
If it's a unschedulable Pod, it will attempt to requeue itself, will not trigger other pods. Can you help me to understand `returns queue for multiple unscheduled pod`, thanks.

Reply:
> Can you help me to understand returns queue for multiple unscheduled pod, thanks.

Sry, I got confused with scheduler_cache update🤦. Updated.